packages/sanity/src/form/form-components-hooks/components.ts

```diff
@@ -0,0 +1,80 @@
+import {SchemaType} from '@sanity/types'
+import React, {useCallback, createElement} from 'react'
+import {PreviewProps} from '../../components/previews'
+import {
+  defaultResolveInputComponent,
+  defaultResolveFieldComponent,
+  defaultResolveItemComponent,
+  defaultResolvePreviewComponent,
+} from '../studio/inputResolver/inputResolver'
+import {InputProps, FieldProps, ItemProps} from '../types'
+
+function useResolveDefaultComponent<T>(props: {
+  componentProps: T & {schemaType: SchemaType}
+  resolver: (schemaType: SchemaType) => React.ComponentType<T>
```

Comment:
Would `resolver` be more specific if it was named `componentResolver`? I guess in this context, it's clear enough.

Reply:
Update the name, good point